src/frontend/next/src/hooks/use-prefer-theme-colors.ts

```diff
@@ -0,0 +1,33 @@
+import { useMemo } from 'react';
+import useMediaQuery from '@material-ui/core/useMediaQuery';
+import { Theme } from '@material-ui/core';
+import { darkTheme, lightTheme } from '../theme';
+
+export default function usePreferredThemeColors(): Theme | null {
+  const isDarkThemePreferred = useMediaQuery('(prefers-color-scheme: dark)');
+  const preferredTheme = useMemo(() => (isDarkThemePreferred ? darkTheme : lightTheme), [
+    isDarkThemePreferred,
+  ]);
+
+  // This portion of code only run on client
+  if (typeof window !== 'undefined') {
+    // First give priority to value stored in localStorage
+    const storedSelectedTheme = localStorage.getItem('theme');
+    if (storedSelectedTheme) {
+      if (storedSelectedTheme === 'dark') {
+        return darkTheme;
+      }
+      return lightTheme;
+    }
+
+    // If there is no value store in local storage then refer to the prefer-color-scheme
+    const osPreferredTheme = isDarkThemePreferred ? 'dark' : 'light';
+
+    // Set the theme value to localStorage for the first time
+    localStorage.setItem('theme', osPreferredTheme);
```

Comment:
What will this do on the server?

Reply:
@humphd : On server there is no localStorage, but this hook still get compile, so this is to avoid what should be running on client not to run on server.

Reply:
thats why I have `if (typeof window !== 'undefined')` to check whether we are on server or not